test/kubernetes_unittest.cc

```diff
@@ -137,4 +147,73 @@ TEST_F(KubernetesTest, ComputePodAssociations) {
       ComputePodAssociations(reader, pod->As<json::Object>());
   EXPECT_EQ(expected_associations->ToString(), associations->ToString());
 }
+
+TEST_F(KubernetesTest, GetContainerMetadata) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "KubernetesClusterLocation: TestClusterLocation\n"
+    "MetadataApiResourceTypeSeparator: \".\"\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+  const auto m = GetContainerMetadata(
+      reader,
+      json::object({
+        {"metadata", json::object({
+          {"namespace", json::string("TestNamespace")},
+          {"name", json::string("TestName")},
+          {"uid", json::string("TestUid")},
+          {"creationTimestamp", json::string("2018-03-03T01:23:45.678901234Z")},
+          {"labels", json::object({{"label", json::string("TestLabel")}})},
+        })}
+      })->As<json::Object>(),
+      json::object({{"name", json::string("TestSpecName")}})->As<json::Object>(),
+      json::object({
+        {"containerID", json::string("docker://TestContainerID")}
+      })->As<json::Object>(),
+      json::string("TestAssociations"),
+      Timestamp(),
+      false);
+
+  EXPECT_EQ(std::vector<std::string>({
+    "k8s_container.TestUid.TestSpecName",
+    "k8s_container.TestNamespace.TestName.TestSpecName",
+    "k8s_container.TestContainerID"}), m.ids());
```

Comment:
Let's break before the `}` and add a trailing comma.

Reply:
Much better, updated.